docker/observation/scripts/setup_elasticsearch_ilm.sh

```
@@ -0,0 +1,83 @@
+MAX_RETRIES=30
+RETRY_INTERVAL=5
+
+echo "Configuring Elasticsearch with ${RETENTION_DAYS}-day retention..."
+
+echo "Waiting for Elasticsearch to be ready..."
+for i in $(seq 1 $MAX_RETRIES); do
+    if curl -s "$ES_HOST/_cluster/health" | grep -q '"status":"green"\|"status":"yellow"'; then
+        echo "Elasticsearch is ready"
+        break
+    fi
+    if [ $i -eq $MAX_RETRIES ]; then
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)